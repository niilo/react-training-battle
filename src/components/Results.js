import React, { Component } from "react";
import { Link } from "react-router-dom";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import Player from "./Player";
import Loading from "./Loading";
import { doFight } from "../results/actions";

class Results extends Component {
  componentDidMount() {
    console.log(this.props);
    const { dispatch } = this.props;
    dispatch(doFight);
  }
  render() {
    const { error, winner, loser, loading } = this.props.results;

    if (loading === true) {
      return <Loading text="downloading" speed={100} />;
    }

    if (error) {
      return (
        <div>
          <p>
            {error}
          </p>
          <Link to="/battle">Reset</Link>
        </div>
      );
    }

    return (
      <div className="row">
        <Player label="Winner" score={winner.score} profile={winner.profile} />
        <Player label="Loser" score={loser.score} profile={loser.profile} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  playerOne: state.battle.playerOne,
  playerTwo: state.battle.playerTwo,
  results: state.results
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      doFight,
      dispatch
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(Results);
