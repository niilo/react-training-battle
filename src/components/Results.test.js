import React from "react";
import Results from "./Results";
import renderer from "react-test-renderer";

it("renders default correctly", () => {
  const tree = renderer.create(<Results />).toJSON();
  expect(tree).toMatchSnapshot();
});
