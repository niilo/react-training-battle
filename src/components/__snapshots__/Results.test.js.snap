// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
<div>
  Results
</div>
`;
